clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1212,6 +1215,14 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-module-file-deps");
   }
 
+  if (!ArgM && Args.hasArg(options::OPT_fintelfpga) &&
```

Comment:
Would it be easier to understand the logic if we make this to an elsif instead? To me, it is, not sure about your view :).

Reply:
Changes I made were based on other usage of `!ArgM` and the fact that the preceding `if` condition was quite large, reducing visibility of what the else would represent.